aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -662,6 +664,29 @@ int64_t _fused_sdp_choice_cpp(const Tensor& query_, const Tensor& key, const Ten
   return static_cast<int64_t>(sdp::SDPBackend::math);
 }
 
+int64_t _fused_sdp_choice_meta(
+    const Tensor& query_,
+    const Tensor& key,
+    const Tensor& value,
+    const c10::optional<Tensor>& attn_mask_,
+    double dropout_p,
+    bool is_causal) {
+  auto query_key_set = query_.key_set();
+  bool has_cuda = query_key_set.has(c10::DispatchKey::CUDA);
```

Comment:
query.device() == "meta" hence the use of the dispatch keys to determine which stub to use